Source/Scene/ClippingPlanesCollection.js

```diff
@@ -0,0 +1,237 @@
+         * @default true
+         */
+        combineClippingRegions : {
+            get : function() {
+                return this._combineClippingRegions;
+            },
+            set : function(value) {
+                if (this._combineClippingRegions !== value) {
+                    this._combineClippingRegions = value;
+                    this._testIntersection = getTestIntersectionFunction(value);
+                }
+            }
+        }
+    });
+
+    function getTestIntersectionFunction(combineClippingRegions) {
+        if (combineClippingRegions) {
+            return function(value) {
+                return (value === Intersect.INSIDE);
+            };
+        }
+
+        return function(value) {
+            return (value === Intersect.OUTSIDE);
+        };
+    }
+
+    var scratchPlane = new Plane(Cartesian3.UNIT_X, 0.0);
+    var scratchMatrix = new Matrix4();
+    /**
+     * Applies the transformations to each plane and packs it into an array.
+     * @private
+     *
+     * @param viewMatrix The 4x4 matrix to transform the plane into eyespace.
+     * @param [array] The array into which the planes will be packed.
+     * @returns {Cartesian4[]} The array of packed planes.
+     */
+    ClippingPlanesCollection.prototype.transformAndPackPlanes = function(viewMatrix, array) {
+        //>>includeStart('debug', pragmas.debug);
+        Check.typeOf.object('viewMatrix', viewMatrix);
+        //>>includeEnd('debug');
+
+        var planes = this.planes;
+        var length = planes.length;
+
+        var i;
+        if (!defined(array)) {
+            array = new Array(length);
+
+            for (i = 0; i < length; ++i) {
+                array[i] = new Cartesian4();
+            }
+        }
+
+        var transform = Matrix4.multiply(viewMatrix, this.modelMatrix, scratchMatrix);
+
+        for (i = 0; i < length; ++i) {
+            var plane = planes[i];
+            var packedPlane = array[i];
+
+            Plane.transform(plane, transform, scratchPlane);
+
+            Cartesian3.clone(scratchPlane.normal, packedPlane);
+            packedPlane.w = scratchPlane.distance;
+        }
+
+        return array;
+    };
+
+    /**
+     * Duplicates this ClippingPlanesCollection instance.
+     *
+     * @param {ClippingPlanesCollection} [result] The object onto which to store the result.
+     * @returns {ClippingPlanesCollection} The modified result parameter or a new ClippingPlanesCollection instance if one was not provided.
+     */
+    ClippingPlanesCollection.prototype.clone = function(result) {
+        if (!defined(result)) {
+            result = new ClippingPlanesCollection();
+        }
+
+        var length = this.planes.length;
+        if (result.planes.length !== length) {
+            result.planes = new Array(length);
+        }
+        for (var i = 0; i < length; ++i) {
+            var plane = this.planes[i];
+            result.planes[i] = new Plane(plane.normal, plane.distance);
```

Comment:
Also try to avoid the `new` here. Above when the new array is created it could initialize with new planes.

Reply:
Is it ok if I point this at the same plane object? The planes aren't manipulated down at the model level after the collection is cloned, just the enabling and modelMatrix. Or better yet just point it at the same array?

Reply:
It's probably better not to use the same plane object or array since callers would expect a deep clone (even though the code right now doesn't care).